src/docs/sdk/unified-api/tracing.mdx

```diff
@@ -0,0 +1,59 @@
+    - Value is expressed in `0.0` - `1.0` float, ranged from 0% to 100% rate
+    - Default makes AM an opt-in feature
+- Introduce `Sentry.startTransaction`
+    - Internally this creates a `Transaction` and returns the instance
+    - Users need to interact with the instance and keep track of it themselves
+    - Internally we can use the `Scope` to hold on to a reference of the current `Transaction` if needed
+- Introduce the new type [Span](/sdk/event-payloads/span/)
+- Introduce new type [Transaction](/sdk/event-payloads/transaction/) which inherits from `Span`
+- `Span` Interface
+    - When a `Span` is created, set the `startTimestamp` to the current time
+    - `SpanContext` is the attribute collection for a `Span` (Can be an implementation detail)
+    - The relation between parent - child is captured in the property `parentSpanId`
+- `Transaction` Interface
+    - A `Transaction` internally holds a flat list of child Spans (not a tree structure)
+    - `Transaction` has additionally a `setName` method the set the name of the transaction
+    - `Transaction` receives a `TransactionContext` on creation (new property vs. `SpanContext` is `name`)
+    - Since a `Transaction` inherits a `Span` it has all functions available and can be interacted with like it was a `Span`
+- `Hub` → Introduce a method called `startSpan`
+    - Creates a new `Span` instance
+    - If there is already a `Span` on the current `Scope`, the created `Span` should be its child
+    - This method deals with sampling, and therefore it should take the `tracesSampleRate` option into account:
+        - If it's a child `Span`, we inherit the decision from the parent
+        - If it's a root span, we need to roll the dice here
+        - Depending on the outcome, the sample decision should be stored in the `Span`'s `sampled` property and again forwarded to its children
+- `Hub` → Introduce a method called `traceHeaders`
+    - This function returns a header (string) `sentry-trace`
+    - The value should be the trace header string of the `Span` that is currently on the `Scope`
+- `Scope` Introduce `setSpan`
+    - This can be used internally to pass a `Span` / `Transaction` around so that integrations can attach children to it
+    - Setting the `transaction` property on the `Scope` (legacy) should overwrite the name of the `Transaction` stored in the `Scope`, if there is one. With that we give users the option to change the transaction name even if they don't have access to the instance of the `Transaction` directly.
+- `Span.finish()`
+    - Just set `endTimestamp` to the current time (in payload `timestamp`)
+- `Transaction.finish()`
+    - `super.finish()` (call finish on Span)
+    - Send it to Sentry only if `sampled == true`
+    - A `Transaction` needs to be wrapped in an `Envelope` and sent to the [Envelope Endpoint](https://develop.sentry.dev/sdk/envelopes/)
+    - The `Transport` should use the same internal queue for `Transactions` / `Events`
```

Comment:
Could you please elaborate on this? Why do we must use the same?
Although likely simpler, it would now require that we have a single max value that now affects tiny transactions and big events. If I understand correctly, errors in a tight loop could fill up the queue and start dropping trace data and break distribute traces.

Reply:
Yep, it's intended, if an SDK is so spammy that the queue can't keep up with sending it, it's fine to drop whatever.

Keep in mind, this is not offline caching.

Reply:
Yes but we could drop the errors that are too spammy but keep the transaction flowing without any problems.
At that point it's possible the errors are getting rate-limited too, while the transactions are not.
If we use a single queue here, it won't matter that we have different rate limits, we'll drop transactions because the internal queue is full of errors.

I just wanted to voice my concern but if you're aware and have decided this, I'll resolve this too.